### DISCUSSION TITLE
Store API Cleanup

### DISCUSSION THREAD

**Discussion by shyim** [shopware/platform#3169] [RFC / Proposal]
**Effort**: low

### Description

Some store API routes returns the entity collection instead of the entity search result. This does not allow to use aggregations.

### Benefits

- Improve functionality of store API
- Unify store API behavior

### Break strategy

We change the API Response of following routes to an entity search result

- https://github.com/shopware/platform/blob/trunk/src/Core/System/Currency/SalesChannel/CurrencyRouteResponse.php
- https://github.com/shopware/platform/blob/trunk/src/Core/Content/Category/SalesChannel/NavigationRouteResponse.php
- https://github.com/shopware/platform/blob/trunk/src/Core/Content/Product/SalesChannel/CrossSelling/ProductCrossSellingRouteResponse.php
- https://github.com/shopware/platform/blob/trunk/src/Core/Content/Sitemap/SalesChannel/SitemapRouteResponse.php

**Comment by ts-mothership** [comment]
Since Shopware 6 has an API first approach, IMHO you should concentrate to improve the full page caching support. For example in https://github.com/shopware/platform/blob/trunk/src/Core/Content/Product/SalesChannel/Listing/ProductListingRoute.php#L88 by supporting more GET requests you can make use of the internal full page cache. If this is not possible, there is a way to convince the FPC with a bit of trickery to cache even POST/Search requests.

The benefit would be high: Having only a minimal bootstrapping for store-api requests would greatly improve loading time. Especially if you need to access multiple endpoints on the first time (loading menu, products, manufacturers, ... on first visit)

[a comment by an automated account is not shown]

**Comment by OliverSkroblin** [comment]
Will be part of the [HTTP Cache rework](https://github.com/shopware/platform/discussions/3299) and in some kind of the read optimized storages (https://github.com/shopware/platform/discussions/3165)
